VariantSurvival_package/R/VariantSurvival.R

```diff
@@ -1,425 +1,403 @@
+        )
+      },height = 900)
+    # 1-year survival time
+    
+    output$table1 <- DT::renderDataTable({
+      svs_gene_input_df <- reactive_no_NAs_metadata()
+      
+      x<- survfit(Surv(time, event)~Phenotype, data = svs_gene_input_df ) %>%
+        tbl_survfit(
+          times = 365.25,
+          label_header = "**1-year survival (95% CI)**"
+        )
+      t <-as_tibble(x)
+      t
+    })
+    # Median survival time
+    output$table2 <- DT::renderDataTable({
+      svs_gene_input_df <- reactive_no_NAs_metadata()
+      
+      x2 <-  survfit(Surv(time, event)~Phenotype, data = svs_gene_input_df ) %>%
+        gtsummary::tbl_survfit(
+          probs = 0.5,
+          label_header = "**Median survival (95% CI)**"
+        )
+      t2 <- as_tibble(x2)
+      t2
+    })
+    #regression table
+    
+    output$table3 <- DT::renderDataTable({
+      svs_gene_input_df <- reactive_no_NAs_metadata()
+      
+      x3 <- coxph(Surv(time, event)~Phenotype, data = svs_gene_input_df ) %>%
+        tbl_regression(exp = TRUE)
+      
+      t3 <-as_tibble(x3)
+      t3
+    })
+  }
+
+  # Run the application
+  shinyApp(ui = ui, server = server)
+}
+
+
+############# Helper functions ###################
+install_load_requirements<- function() {
+  if (!require("shiny")) install.packages("shiny")
+  if (!require("shinyWidgets")) install.packages("shinyWidgets")
+  if (!require("shinythemes")) install.packages("shinythemes")
+  if (!require("shinycssloaders")) install.packages("shinycssloaders")
+  if (!require("shinydashboard")) install.packages("shinydashboard")
+  if (!require("DT")) install.packages("DT")
+  if (!require("dplyr")) install.packages("dplyr")
+  if (!require("tidyverse")) install.packages("tidyverse")
+  if (!require("vcfR")) install.packages("vcfR")
+  if (!require("readr")) install.packages("readr")
+  if (!require("ggsurvfit")) install.packages("ggsurvfit")
+  if (!require("survival")) install.packages("survival")
+  if (!require("ggplot2")) install.packages("ggplot2")
+  if (!require("survminer")) install.packages("survminer")
+  if (!require("lubridate")) install.packages("lubridate")
+  if (!require("gtsummary")) install.packages("gtsummary")
+  if (!require("tibble")) install.packages("tibble")
+  library(shiny)
+  library(shinydashboard)
+  library(shinythemes)
+  library(shinyWidgets)
+  library(shinycssloaders)
+  library(DT)
+  library(dplyr)
+  library(tidyverse)
+  library(vcfR)
+  library(readr)
+  library(readxl)
+  library(ggsurvfit)
+  library(survival)
+  library(ggplot2)
+  library(survminer)
+  library(lubridate)
+  library(gtsummary)
+  library(tibble)
+}
+
+
+#' `get_disease_gene_list` parses an existing excel 
+#' file containing all known target genes associated
+#' with the selected disease.
+#' @param input_disease: input disease selection
+#' @return  genes_list
+get_disease_gene_list <- function(input_disease) {
+  disease_gene_select <- disease_gene[,input$disease_n]
```

Comment:
what is disease_gene? you're referencing a df that's not defined within the function. Better pass it as an argument.

Reply:
disease_gene is dataframe = excel file containing genes list
L17: disease_gene <- read_excel("disease_gene.xlsx")
the command disease_gene[,input$disease_n] should extract the column = input$disease_n